core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -1755,7 +1755,7 @@ public RelBuilder aggregate(GroupKey groupKey, Iterable<AggCall> aggCalls) {
           newProjects.add(project.getProjects().get(i));
           builder.add(project.getRowType().getFieldList().get(i));
         }
-        r = project.copy(r.getTraitSet(), project.getInput(), newProjects,
+        r = project.copy(cluster.traitSet(), project.getInput(), newProjects,
             builder.build());
```

Comment:
Can we try to make the trait set correct in current code ?

Reply:
Not worth it. The bug has been there for years. Logical operator should not have any traitset, including LogicalSort.

Reply:
Is LogicalSort useless then?

Reply:
We should not have LogicalSort. It also incorporates the functionality of LIMIT, another wrong design.